3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js

```diff
@@ -20,18 +20,62 @@ Use async/await and try/catch to handle promises.
 Try and avoid using global variables. As much as possible, try and use function 
 parameters and return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+
+async function fetchData(url) {
+  const fetchData = await fetch(url);
+  const data = await fetchData.json();
+  return data.results;
 }
 
-function fetchAndPopulatePokemons(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function fetchAndPopulatePokemons(array) {
+  let index = 0;
+  array.forEach((pokemon) => {
+    const option = document.createElement('option');
+    option.textContent = pokemon.name;
+    option.value = index;
+    const select = document.querySelector('select');
+    select.appendChild(option);
+    index++;
+  });
 }
 
-function fetchImage(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+async function fetchImage(url) {
+  const pokemonUrl = await fetch(url);
+  const pokemonData = await pokemonUrl.json();
+  const imgSrc = pokemonData.sprites.back_default;
+  const image = document.querySelector('img');
+  image.src = imgSrc;
 }
 
-function main() {
-  // TODO complete this function
+async function main() {
+  try {
+    const body = document.body;
+    const select = document.createElement('select');
+    const btn = document.createElement('button');
+    btn.type = 'submit';
+    body.appendChild(select);
+    body.appendChild(btn);
+    btn.textContent = 'Get Pokemon';
+
+    btn.addEventListener('click', async () => {
+      const pokemonArray = await fetchData(
+        'https://pokeapi.co/api/v2/pokemon?limit=151'
+      );
+      fetchAndPopulatePokemons(pokemonArray);
+    });
+
+    select.addEventListener('change', async (e) => {
+      const pokemonArray = await fetchData(
+        'https://pokeapi.co/api/v2/pokemon?limit=151'
+      );
+      const image = document.createElement('img');
+      document.body.appendChild(image);
+      const pokemonUrl = pokemonArray[e.target.value].url;
```

Comment:
```suggestion
      const pokemonUrl = e.target.value;
```

The value _is_ now the URL.